app/code/Magento/Newsletter/Controller/Adminhtml/Problem/Index.php

```diff
@@ -6,9 +6,10 @@
  */
 namespace Magento\Newsletter\Controller\Adminhtml\Problem;
 
+use Magento\Framework\App\Action\HttpPostActionInterface as HttpPostActionInterface;
 use Magento\Framework\App\Action\HttpGetActionInterface as HttpGetActionInterface;
 
-class Index extends \Magento\Newsletter\Controller\Adminhtml\Problem implements HttpGetActionInterface
+class Index extends \Magento\Newsletter\Controller\Adminhtml\Problem implements HttpGetActionInterface, HttpPostActionInterface
```

Comment:
Hi @GovindaSharma. Thanks for collaboration. Can you please explain how your changes related to the bug in the issue?

Reply:
Hello @VladimirZaets ,as when grid is loading then it is get request but when you click on Search or Rest Filter button it becomes post request,but in controller only HttpGetActionInterface is implemented because of this filters not working,so i added the HttpPostActionInterface ,so whenever filters is called it is handled by HttpPostActionInterface.

You can check in Newsletter queue Grid,there it is working fine because in the controller(app/code/Magento/Newsletter/Controller/Adminhtml/Queue/Index.php) both HttpPostActionInterface and HttpGetActionInterface is implemented.

Reply:
@GovindaSharma Hi, I understand the case. So, the code that currently exists in Magento will be refactored (or you can do it also in another PR if you want).

The right way to resolve this issue is creating additional POST controller and use it for filters operations.

Reply:
@VladimirZaets ok then i will change accordingly,since controller is already created we just need to pass that controller url.I will update you on this.
Thank You